Hadoop tests fail with OpenJ9-17 due to 'No LoginModule found for com.ibm.security.auth.module.JAASLoginModule' [HZ-980]

_master_ (commit 8dbea9097209242f5cc6e27d8b433cd7b997e545)

Failed on OpenJ9-17:
- https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-17/1/testReport/com.hazelcast.jet.hadoop.file/
- https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-17/1/testReport/com.hazelcast.jet.hadoop.impl/
- http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-17/lastCompletedBuild/testReport/com.hazelcast.jet.sql.impl.connector.file/SqlHadoopTest/com_hazelcast_jet_sql_impl_connector_file_SqlHadoopTest/

<details><summary>Stacktrace (for 'com.hazelcast.jet.hadoop.file.CsvFileFormatTest.shouldReadCsvFileToObject[0: useHadoop=true]' as an example):</summary>

```
java.util.concurrent.CompletionException: com.hazelcast.jet.JetException: java.lang.RuntimeException: org.apache.hadoop.security.KerberosAuthException: failure to login: javax.security.auth.login.LoginException: No LoginModule found for com.ibm.security.auth.module.JAASLoginModule
	at java.base/java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:413)
	at java.base/java.util.concurrent.CompletableFuture.join(CompletableFuture.java:2118)
	at com.hazelcast.jet.Job.join(Job.java:78)
	at com.hazelcast.jet.hadoop.file.BaseFileFormatTest.assertItemsInSource(BaseFileFormatTest.java:88)
	at com.hazelcast.jet.hadoop.file.BaseFileFormatTest.assertItemsInSource(BaseFileFormatTest.java:71)
	at com.hazelcast.jet.hadoop.file.BaseFileFormatTest.assertItemsInSource(BaseFileFormatTest.java:65)
	at com.hazelcast.jet.hadoop.file.CsvFileFormatTest.shouldReadCsvFileToObject(CsvFileFormatTest.java:76)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37)
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104)
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:63)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:39)
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:884)
Caused by: com.hazelcast.jet.JetException: java.lang.RuntimeException: org.apache.hadoop.security.KerberosAuthException: failure to login: javax.security.auth.login.LoginException: No LoginModule found for com.ibm.security.auth.module.JAASLoginModule
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$joinSubmittedJob$9(JobCoordinationService.java:452)
	at com.hazelcast.jet.impl.JobCoordinationService$$Lambda$1473/0x00000000d40a8920.apply(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:911)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
	at com.hazelcast.jet.impl.util.NonCompletableFuture.internalCompleteExceptionally(NonCompletableFuture.java:72)
	at com.hazelcast.jet.impl.MasterJobContext.setFinalResult(MasterJobContext.java:500)
	at com.hazelcast.jet.impl.MasterJobContext.lambda$finalizeJob$21(MasterJobContext.java:688)
	at com.hazelcast.jet.impl.MasterJobContext$$Lambda$1528/0x00000000d41c1850.accept(Unknown Source)
	at com.hazelcast.jet.impl.util.ExceptionUtil.lambda$withTryCatch$0(ExceptionUtil.java:172)
	at com.hazelcast.jet.impl.util.ExceptionUtil$$Lambda$1397/0x0000000004ff80e8.accept(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:887)
	at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2325)
	at com.hazelcast.jet.impl.MasterJobContext.lambda$finalizeJob$22(MasterJobContext.java:684)
	at com.hazelcast.jet.impl.MasterJobContext$$Lambda$1507/0x00000000d41b3e18.run(Unknown Source)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$51(JobCoordinationService.java:1267)
	at com.hazelcast.jet.impl.JobCoordinationService$$Lambda$1394/0x0000000004ff5200.call(Unknown Source)
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1276)
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1266)
	at com.hazelcast.jet.impl.MasterJobContext.finalizeJob(MasterJobContext.java:635)
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$2(MasterJobContext.java:241)
	at com.hazelcast.jet.impl.MasterJobContext$$Lambda$1466/0x00000000d40a7a18.run(Unknown Source)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$51(JobCoordinationService.java:1267)
	at com.hazelcast.jet.impl.JobCoordinationService$$Lambda$1394/0x0000000004ff5200.call(Unknown Source)
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1276)
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1266)
	at com.hazelcast.jet.impl.MasterJobContext.tryStartJob(MasterJobContext.java:198)
	at com.hazelcast.jet.impl.JobCoordinationService.tryStartJob(JobCoordinationService.java:1163)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitJob$1(JobCoordinationService.java:309)
	at com.hazelcast.jet.impl.JobCoordinationService$$Lambda$1393/0x0000000004ff4da0.run(Unknown Source)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$51(JobCoordinationService.java:1267)
	at com.hazelcast.jet.impl.JobCoordinationService$$Lambda$1394/0x0000000004ff5200.call(Unknown Source)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$52(JobCoordinationService.java:1288)
	at com.hazelcast.jet.impl.JobCoordinationService$$Lambda$1395/0x0000000004ff5458.call(Unknown Source)
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64)
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:884)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)
Caused by: java.lang.RuntimeException: org.apache.hadoop.security.KerberosAuthException: failure to login: javax.security.auth.login.LoginException: No LoginModule found for com.ibm.security.auth.module.JAASLoginModule
	at org.apache.hadoop.mapreduce.task.JobContextImpl.<init>(JobContextImpl.java:74)
	at org.apache.hadoop.mapreduce.Job.<init>(Job.java:152)
	at org.apache.hadoop.mapreduce.Job.getInstance(Job.java:195)
	at com.hazelcast.jet.hadoop.impl.HadoopFileSourceFactory$1.acceptEx(HadoopFileSourceFactory.java:141)
	at com.hazelcast.jet.hadoop.impl.HadoopFileSourceFactory$1.acceptEx(HadoopFileSourceFactory.java:127)
	at com.hazelcast.function.ConsumerEx.accept(ConsumerEx.java:47)
	at com.hazelcast.jet.hadoop.impl.ReadHadoopNewApiP$MetaSupplier.updateConfiguration(ReadHadoopNewApiP.java:223)
	at com.hazelcast.jet.hadoop.impl.ReadHadoopNewApiP$MetaSupplier.init(ReadHadoopNewApiP.java:192)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlanBuilder.lambda$createExecutionPlans$8d3991c6$1(ExecutionPlanBuilder.java:107)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlanBuilder$$Lambda$1499/0x00000000d40bc820.runEx(Unknown Source)
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:31)
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:519)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlanBuilder.createExecutionPlans(ExecutionPlanBuilder.java:106)
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$ab649be7$1(MasterJobContext.java:229)
	at com.hazelcast.jet.impl.MasterJobContext$$Lambda$1481/0x00000000d40b42e0.runEx(Unknown Source)
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:31)
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:519)
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$2(MasterJobContext.java:228)
	... 20 more
Caused by: org.apache.hadoop.security.KerberosAuthException: failure to login: javax.security.auth.login.LoginException: No LoginModule found for com.ibm.security.auth.module.JAASLoginModule
	at org.apache.hadoop.security.UserGroupInformation.doSubjectLogin(UserGroupInformation.java:1986)
	at org.apache.hadoop.security.UserGroupInformation.createLoginUser(UserGroupInformation.java:719)
	at org.apache.hadoop.security.UserGroupInformation.getLoginUser(UserGroupInformation.java:669)
	at org.apache.hadoop.security.UserGroupInformation.getCurrentUser(UserGroupInformation.java:579)
	at org.apache.hadoop.mapreduce.task.JobContextImpl.<init>(JobContextImpl.java:72)
	... 37 more
Caused by: javax.security.auth.login.LoginException: No LoginModule found for com.ibm.security.auth.module.JAASLoginModule
	at java.base/javax.security.auth.login.LoginContext.invoke(LoginContext.java:739)
	at java.base/javax.security.auth.login.LoginContext$4.run(LoginContext.java:679)
	at java.base/javax.security.auth.login.LoginContext$4.run(LoginContext.java:677)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:775)
	at java.base/javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:677)
	at java.base/javax.security.auth.login.LoginContext.login(LoginContext.java:587)
	at org.apache.hadoop.security.UserGroupInformation$HadoopLoginContext.login(UserGroupInformation.java:2065)
	at org.apache.hadoop.security.UserGroupInformation.doSubjectLogin(UserGroupInformation.java:1975)
	... 41 more
```

</details>

<details><summary>Standard output (for 'com.hazelcast.jet.hadoop.file.CsvFileFormatTest.shouldReadCsvFileToObject[0: useHadoop=true]' as an example):</summary>

```
17:23:35,379  INFO || - [MetricsConfigHelper] Time-limited test - [LOCAL] [dev] [5.1-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
17:23:35,380  INFO || - [logo] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
17:23:35,380  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
17:23:35,380  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Hazelcast Platform 5.1-SNAPSHOT (20220217 - 8dbea90) starting at [127.0.0.1]:5701
17:23:35,380  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Cluster name: dev
17:23:35,380  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed.
To enable integrity checker do one of the following: 
  - Change member config using Java API: config.setIntegrityCheckerEnabled(true);;
  - Change XML/YAML configuration property: Set hazelcast.integrity-checker.enabled to true
  - Add system property: -Dhz.integritychecker.enabled=true
  - Add environment variable: HZ_INTEGRITYCHECKER_ENABLED=true (recommended when running container/docker image)
17:23:35,380  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Jet is enabled
17:23:35,383  INFO || - [MetricsConfigHelper] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
17:23:35,387  WARN || - [CPSubsystem] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
17:23:35,391  INFO || - [JetServiceBackend] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
17:23:35,392  INFO || - [Diagnostics] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
17:23:35,392  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] [127.0.0.1]:5701 is STARTING
17:23:35,393  INFO || - [ClusterService] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5701 - a39b25a3-b6a7-4c89-a867-3ffafbe3c657 this
]

17:23:35,393  INFO || - [JobCoordinationService] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Jet started scanning for jobs
17:23:35,393  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] [127.0.0.1]:5701 is STARTED
17:23:35,393 DEBUG || - [JobCoordinationService] hz.frosty_wiles.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
17:23:35,399  INFO || - [PartitionStateManager] hz.frosty_wiles.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Initializing cluster partition table arrangement...
17:23:35,400 DEBUG || - [JobClassLoaderService] hz.frosty_wiles.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Creating job classLoader for job 0796-6c57-d840-0001
17:23:35,400 DEBUG || - [JobClassLoaderService] hz.frosty_wiles.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Create processor classloader map for job 0796-6c57-d840-0001
17:23:35,409 DEBUG || - [Planner] hz.frosty_wiles.cached.thread-1 - Watermarks in the pipeline will be throttled to 1000
17:23:35,427  INFO || - [JobCoordinationService] hz.frosty_wiles.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Starting job 0796-6c57-d840-0001 based on submit request
17:23:35,438  INFO || - [MasterJobContext] hz.frosty_wiles.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Didn't find any snapshot to restore for job '0796-6c57-d840-0001', execution 0796-6c57-d841-0001
17:23:35,438  INFO || - [MasterJobContext] hz.frosty_wiles.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Start executing job '0796-6c57-d840-0001', execution 0796-6c57-d841-0001, execution graph in DOT format:
digraph DAG {
	"files(path=/home/jenkins/jenkins_slave/workspace/Hazelcast-master-Openj9-17/extensions/hadoop/src/test/resources, glob=file.csv, hadoop=true" [localParallelism=2];
	"assertCollected" [localParallelism=1];
	"files(path=/home/jenkins/jenkins_slave/workspace/Hazelcast-master-Openj9-17/extensions/hadoop/src/test/resources, glob=file.csv, hadoop=true" -> "assertCollected" [label="distributed-partitioned", queueSize=1024];
}
HINT: You can use graphviz or http://viz-js.com/ to visualize the printed graph.
17:23:35,438 DEBUG || - [MasterJobContext] hz.frosty_wiles.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Building execution plan for job '0796-6c57-d840-0001', execution 0796-6c57-d841-0001
17:23:35,439 DEBUG || - [JobClassLoaderService] hz.frosty_wiles.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Finish JobClassLoaders phaseCount = 0, removing classloaders for jobId=0796-6c57-d840-0001
17:23:35,439 ERROR || - [MasterJobContext] hz.frosty_wiles.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Execution of job '0796-6c57-d840-0001', execution 0796-6c57-d841-0001 failed
	Start time: 2022-02-17T17:23:35.428
	Duration: 00:00:00.011
	To see additional job metrics enable JobConfig.storeMetricsAfterJobCompletion
java.lang.RuntimeException: org.apache.hadoop.security.KerberosAuthException: failure to login: javax.security.auth.login.LoginException: No LoginModule found for com.ibm.security.auth.module.JAASLoginModule
	at org.apache.hadoop.mapreduce.task.JobContextImpl.<init>(JobContextImpl.java:74) ~[hadoop-mapreduce-client-core-3.3.1.jar:?]
	at org.apache.hadoop.mapreduce.Job.<init>(Job.java:152) ~[hadoop-mapreduce-client-core-3.3.1.jar:?]
	at org.apache.hadoop.mapreduce.Job.getInstance(Job.java:195) ~[hadoop-mapreduce-client-core-3.3.1.jar:?]
	at com.hazelcast.jet.hadoop.impl.HadoopFileSourceFactory$1.acceptEx(HadoopFileSourceFactory.java:141) ~[classes/:?]
	at com.hazelcast.jet.hadoop.impl.HadoopFileSourceFactory$1.acceptEx(HadoopFileSourceFactory.java:127) ~[classes/:?]
	at com.hazelcast.function.ConsumerEx.accept(ConsumerEx.java:47) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.hadoop.impl.ReadHadoopNewApiP$MetaSupplier.updateConfiguration(ReadHadoopNewApiP.java:223) ~[classes/:?]
	at com.hazelcast.jet.hadoop.impl.ReadHadoopNewApiP$MetaSupplier.init(ReadHadoopNewApiP.java:192) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlanBuilder.lambda$createExecutionPlans$8d3991c6$1(ExecutionPlanBuilder.java:107) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlanBuilder$$Lambda$1499/0x00000000d40bc820.runEx(Unknown Source) ~[?:?]
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:31) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:519) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlanBuilder.createExecutionPlans(ExecutionPlanBuilder.java:106) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$ab649be7$1(MasterJobContext.java:229) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterJobContext$$Lambda$1481/0x00000000d40b42e0.runEx(Unknown Source) ~[?:?]
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:31) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:519) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$2(MasterJobContext.java:228) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterJobContext$$Lambda$1466/0x00000000d40a7a18.run(Unknown Source) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$51(JobCoordinationService.java:1267) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService$$Lambda$1394/0x0000000004ff5200.call(Unknown Source) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1276) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1266) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterJobContext.tryStartJob(MasterJobContext.java:198) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService.tryStartJob(JobCoordinationService.java:1163) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitJob$1(JobCoordinationService.java:309) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService$$Lambda$1393/0x0000000004ff4da0.run(Unknown Source) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$51(JobCoordinationService.java:1267) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService$$Lambda$1394/0x0000000004ff5200.call(Unknown Source) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$52(JobCoordinationService.java:1288) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService$$Lambda$1395/0x0000000004ff5458.call(Unknown Source) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:884) [?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
Caused by: org.apache.hadoop.security.KerberosAuthException: failure to login: javax.security.auth.login.LoginException: No LoginModule found for com.ibm.security.auth.module.JAASLoginModule
	at org.apache.hadoop.security.UserGroupInformation.doSubjectLogin(UserGroupInformation.java:1986) ~[hadoop-common-3.3.1.jar:?]
	at org.apache.hadoop.security.UserGroupInformation.createLoginUser(UserGroupInformation.java:719) ~[hadoop-common-3.3.1.jar:?]
	at org.apache.hadoop.security.UserGroupInformation.getLoginUser(UserGroupInformation.java:669) ~[hadoop-common-3.3.1.jar:?]
	at org.apache.hadoop.security.UserGroupInformation.getCurrentUser(UserGroupInformation.java:579) ~[hadoop-common-3.3.1.jar:?]
	at org.apache.hadoop.mapreduce.task.JobContextImpl.<init>(JobContextImpl.java:72) ~[hadoop-mapreduce-client-core-3.3.1.jar:?]
	... 37 more
Caused by: javax.security.auth.login.LoginException: No LoginModule found for com.ibm.security.auth.module.JAASLoginModule
	at javax.security.auth.login.LoginContext.invoke(LoginContext.java:739) ~[?:?]
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:679) ~[?:?]
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:677) ~[?:?]
	at java.security.AccessController.doPrivileged(AccessController.java:775) ~[?:?]
	at javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:677) ~[?:?]
	at javax.security.auth.login.LoginContext.login(LoginContext.java:587) ~[?:?]
	at org.apache.hadoop.security.UserGroupInformation$HadoopLoginContext.login(UserGroupInformation.java:2065) ~[hadoop-common-3.3.1.jar:?]
	at org.apache.hadoop.security.UserGroupInformation.doSubjectLogin(UserGroupInformation.java:1975) ~[hadoop-common-3.3.1.jar:?]
	at org.apache.hadoop.security.UserGroupInformation.createLoginUser(UserGroupInformation.java:719) ~[hadoop-common-3.3.1.jar:?]
	at org.apache.hadoop.security.UserGroupInformation.getLoginUser(UserGroupInformation.java:669) ~[hadoop-common-3.3.1.jar:?]
	at org.apache.hadoop.security.UserGroupInformation.getCurrentUser(UserGroupInformation.java:579) ~[hadoop-common-3.3.1.jar:?]
	at org.apache.hadoop.mapreduce.task.JobContextImpl.<init>(JobContextImpl.java:72) ~[hadoop-mapreduce-client-core-3.3.1.jar:?]
	... 37 more
17:23:35,441 DEBUG || - [JobCoordinationService] hz.frosty_wiles.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] job '0796-6c57-d840-0001', execution 0796-6c57-d841-0001 is completed
17:23:35,441  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
17:23:35,442 DEBUG || - [JobCoordinationService] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Added a shutting-down member: a39b25a3-b6a7-4c89-a867-3ffafbe3c657
17:23:35,442  INFO || - [MigrationManager] hz.frosty_wiles.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Shutdown request of Member [127.0.0.1]:5701 - a39b25a3-b6a7-4c89-a867-3ffafbe3c657 this is handled
17:23:35,442  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Shutting down connection manager...
17:23:35,442  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Shutting down node engine...
17:23:35,444  INFO || - [NodeExtension] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Destroying node NodeExtension.
17:23:35,444  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Hazelcast Shutdown is completed in 3 ms.
17:23:35,444  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
17:23:35,444  INFO || - [LifecycleService] Thread-141 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
17:23:35,445  INFO || - [Node] Thread-141 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Node is already shutting down... Waiting for shutdown process to complete...
17:23:35,445  INFO || - [LifecycleService] Thread-141 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
BuildInfo right after shouldReadCsvFileToObject[0: useHadoop=true](com.hazelcast.jet.hadoop.file.CsvFileFormatTest): BuildInfo{version='5.1-SNAPSHOT', build='20220217', buildNumber=20220217, revision=8dbea90, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'shouldReadCsvFileToObject[0: useHadoop=true](com.hazelcast.jet.hadoop.file.CsvFileFormatTest):'
17:23:35, accumulated pauses: 2 ms, max pause: 0 ms, pauses over 1000 ms: 0
```

</details>

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/file/SqlHadoopTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/file/SqlHadoopTest.java
@@ -57,6 +57,11 @@ public class SqlHadoopTest extends SqlTestSupport {
     public static void setUpClass() throws Exception {
         assumeThatNoWindowsOS();
 
+        // Tests will fail on IBM JDK17 with error:
+        // No LoginModule found for com.ibm.security.auth.module.JAASLoginModule
+        // see https://github.com/hazelcast/hazelcast/issues/20754
+        assumeThatNotIBMJDK17();
+
         initialize(1, null);
         sqlService = instance().getSql();
 

